src/engine-actions/get-settled-swap-preimage.js

```diff
@@ -1,85 +1,41 @@
+ * @throws {Error} If the invoice is not in a SETTLED state
  */
-function getSettledSwapPreimage (swapHash) {
-  return new Promise(async (resolve, reject) => {
-    try {
-      if (!swapHash) {
-        throw new Error('Swaphash must be defined')
-      }
-
-      // We convert the swapHash from base64 string to hex because lookupInvoice takes in a hex
-      // encoded rhash as an argument.
-      const paymentHash = Buffer.from(swapHash, 'base64').toString('hex')
-
-      this.logger.debug('Looking up invoice by paymentHash:', { paymentHash })
-      // Before subscribing to invoices we lookup to see if it has already been settled,
-      // if so, we can return immediately to the caller
-      const { settled, rPreimage } = await lookupInvoice(paymentHash, { client: this.client })
-
-      this.logger.debug('Invoice has been retrieved: ', { settled })
-
-      if (settled) {
-        return resolve(rPreimage)
-      }
-
-      const subscription = subscribeInvoices({ client: this.client })
-
-      const errorListener = (err) => {
-        this.logger.error(`Error from subscribeInvoices stream`, err)
-        return finish(err)
-      }
-
-      const endListener = () => {
-        this.logger.error(`LND closed subscribeInvoices stream before returning our value`)
-        return finish(new Error(`LND closed stream to retrieve invoice`))
-      }
+async function getSettledSwapPreimage (swapHash) {
+  if (!swapHash) {
+    throw new Error('Swap hash must be defined')
+  }
 
-      const dataListener = (response) => {
-        const { memo, rHash, settled, settleDate, rPreimage } = response
-        this.logger.debug('Received invoice subscription from LND', { memo, rHash, settled })
-        if (rHash === swapHash) {
-          this.logger.debug(`Invoice subscription matched our observed swapHash`, { swapHash, memo })
+  // We convert the swapHash from base64 string to hex because lookupInvoice takes in a hex
+  // encoded rhash as an argument.
+  const paymentHash = Buffer.from(swapHash, 'base64').toString('hex')
 
-          if (settled) {
-            this.logger.debug(`Invoice is settled as of ${settleDate}`)
-            return finish(null, rPreimage)
-          }
-        }
-      }
+  this.logger.debug('Looking up invoice by paymentHash:', { paymentHash })
 
-      // Helper to make sure we tear down our listeners
-      const finish = (err, response) => {
-        subscription.removeListener('error', errorListener)
-        subscription.removeListener('end', endListener)
-        subscription.removeListener('data', dataListener)
+  const { state, rPreimage } = await lookupInvoice(paymentHash, { client: this.client })
 
-        if (err) {
-          return reject(err)
-        }
+  this.logger.debug('Invoice has been retrieved: ', { state })
 
-        resolve(response)
-      }
+  if (state !== INVOICE_STATES.SETTLED) {
+    throw new Error(`Cannot retrieve preimage from an invoice in a ${state} state.`)
```

Comment:
need to update this error for the new usage

Reply:
What do you mean? I think it's already updated?

Reply:
I read this wrong. ignore the comment